go/libraries/doltcore/sqle/dtables/ignore_table.go

```diff
@@ -0,0 +1,275 @@
+// for each row, followed by a call to Close() when all rows have been processed.
+func (it *IgnoreTable) Replacer(ctx *sql.Context) sql.RowReplacer {
+	return newIgnoreWriter(it)
+}
+
+// Updater returns a RowUpdater for this table. The RowUpdater will have Update called once for each row to be
+// updated, followed by a call to Close() when all rows have been processed.
+func (it *IgnoreTable) Updater(ctx *sql.Context) sql.RowUpdater {
+	return newIgnoreWriter(it)
+}
+
+// Inserter returns an Inserter for this table. The Inserter will get one call to Insert() for each row to be
+// inserted, and will end with a call to Close() to finalize the insert operation.
+func (it *IgnoreTable) Inserter(*sql.Context) sql.RowInserter {
+	return newIgnoreWriter(it)
+}
+
+// Deleter returns a RowDeleter for this table. The RowDeleter will get one call to Delete for each row to be deleted,
+// and will end with a call to Close() to finalize the delete operation.
+func (it *IgnoreTable) Deleter(*sql.Context) sql.RowDeleter {
+	return newIgnoreWriter(it)
+}
+
+var _ sql.RowReplacer = (*ignoreWriter)(nil)
+var _ sql.RowUpdater = (*ignoreWriter)(nil)
+var _ sql.RowInserter = (*ignoreWriter)(nil)
+var _ sql.RowDeleter = (*ignoreWriter)(nil)
+
+type ignoreWriter struct {
+	it                      *IgnoreTable
+	errDuringStatementBegin error
+	workingSet              *doltdb.WorkingSet
+	prevHash                *hash.Hash
+	tableWriter             writer.TableWriter
+}
+
+func newIgnoreWriter(it *IgnoreTable) *ignoreWriter {
+	return &ignoreWriter{it, nil, nil, nil, nil}
+}
+
+// Insert inserts the row given, returning an error if it cannot. Insert will be called once for each row to process
+// for the insert operation, which may involve many rows. After all rows in an operation have been processed, Close
+// is called.
+func (iw *ignoreWriter) Insert(ctx *sql.Context, r sql.Row) error {
+	if err := iw.errDuringStatementBegin; err != nil {
+		return err
+	}
+	return iw.tableWriter.Insert(ctx, r)
+}
+
+// Update the given row. Provides both the old and new rows.
+func (iw *ignoreWriter) Update(ctx *sql.Context, old sql.Row, new sql.Row) error {
+	if err := iw.errDuringStatementBegin; err != nil {
+		return err
+	}
+	return iw.tableWriter.Update(ctx, old, new)
+}
+
+// Delete deletes the given row. Returns ErrDeleteRowNotFound if the row was not found. Delete will be called once for
+// each row to process for the delete operation, which may involve many rows. After all rows have been processed,
+// Close is called.
+func (iw *ignoreWriter) Delete(ctx *sql.Context, r sql.Row) error {
+	if err := iw.errDuringStatementBegin; err != nil {
+		return err
+	}
+	return iw.tableWriter.Delete(ctx, r)
+}
+
+// StatementBegin is called before the first operation of a statement. Integrators should mark the state of the data
+// in some way that it may be returned to in the case of an error.
+func (iw *ignoreWriter) StatementBegin(ctx *sql.Context) {
```

Comment:
It feels like we should be able to make this code simpler... I like the experience and design you're going after, just wondering if there's an easier way to implement it...  I'll dig around a little more and see if I can come up with any helpful ideas.

Reply:
I still think there's probably a way to simplify this but I haven't come up with any good ideas yet, are you okay with submitting it as is so we can get the feature out?

Reply:
Yeah, I don't have a better suggestion for ya, and we can always come back and clean up these internals, so just rolling forward and getting this out sounds great.